source/models/index.py

```diff
@@ -0,0 +1,42 @@
+import os
+from outputs.exceptions import *
+
+class Index:
+    def __init__(self, table, index_value):
+        self.table = table
+        self.index_value = index_value
+
+    def serialize(self):
+        os.makedirs(self.get_path(), exist_ok = True)
+    
+    def get_path(self):
+        return os.path.join(self.table.get_path(), os.path.join("indices" , self.index_value))
+    
+    def add_primary_key(self, primary_key, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
+        old_primary_keys = set(self.get_primary_key(index_value))                
+        old_primary_keys.add(primary_key)
+        Index.__write_in_file(path = index_value_file_path, data = old_primary_keys)
+    
+    def delete_primary_key(self, primary_key, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
+        old_primary_keys = self.get_primary_key(index_value)
+        old_primary_keys.remove(primary_key)
+        Index.__write_in_file(path = index_value_file_path, data = old_primary_keys)
+
+    def get_primary_key(self, index_value):
+        index_value_file_path = os.path.join(self.get_path(), index_value + ".txt")
```

Comment:
You should have method called `get_path(index_value)`
You will use it later in get or set

Reply:
```index_value``` is passed to the self in the ```init``` so I think I should pass the ```self``` to the get pass instead.
and also in ```index.py``` a lot of functions take ```index_value```  as a parameter beside the ```self```` I should not pass it as a parameter and take it from the self, shouldn't I ??